rel/value_tuple.go

```diff
@@ -122,6 +122,15 @@ func NewXML(tag []rune, attrs []Attr, children ...Value) Tuple {
 	return EmptyTuple.With("@xml", b.Finish())
 }
 
+func (t *GenericTuple) Canonical() Tuple {
```

Comment:
Would it be worth making this part of the `Tuple` interface
pro: can be called on any tuple without knowing if it is specific or not.
con: basically a noop for anything thats not a GenericTuple.

IMO its not worth it, the type cast and `ok` check is good enough. Just putting the idea out there.

Reply:
Would be worth considering but for the fact that it's temporary hack.